src/common.jl

```diff
@@ -14,6 +14,33 @@ function get_codes(gnss::AbstractGNSS{T}) where T <: CuMatrix
     gnss.codes
 end
 
+"""
+$(SIGNATURES)
+Generate a code signal for PRN-Number `prn` of system `gnss` at chip rate `code_frequency`,
+sampled at sampling rate `sampling_frequency`
+"""
+function gen_code!(
+    code::AbstractVector,
+    gnss::AbstractGNSS,
+    prn::Integer,
+    sampling_frequency::Frequency,
+    code_frequency::Frequency = get_code_frequency(gnss),
+    start_phase = 0.0
+)
+    num_samples = length(code)
+    fixed_point = sizeof(Int) * 8 - 1 - min_bits_for_code_length(gnss)
+    FP = Fixed{Int, fixed_point}
+    total_code_length = FP(get_code_length(gnss) * get_secondary_code_length(gnss))
+    delta = FP(code_frequency / sampling_frequency)
+    code_phase = FP(mod(start_phase, total_code_length))
+    @inbounds for i ∈ 1:num_samples
+        code[i] = get_code_unsafe(gnss, code_phase, prn)
+        code_phase += delta
+        code_phase -= (code_phase >= total_code_length) * total_code_length
```

Comment:
Maybe we should think about adding another bit for the integer part of `code_phase`, as currently `delta>1` leads to an undetected overflow with `GPSL1()` signals. On the other hand, `delta>1` is not really useful...

Reply:
I think we could throw an error if `delta > 1`. Something like:
```
delta > 1 && error("The sampling frequency should be higher than the code frequency")
```

Reply:
:+1: